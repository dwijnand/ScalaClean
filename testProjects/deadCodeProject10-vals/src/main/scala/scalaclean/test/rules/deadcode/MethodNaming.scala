
trait t1 {
  def m1: Int
}

trait t2 extends t1{
  val m1: Int
}

trait t3 extends t1{
  val m1: Int = 1
}

abstract class c1 extends t1{
  val m1: Int
}

class c2 extends t1{
  val m1: Int = 7
}


