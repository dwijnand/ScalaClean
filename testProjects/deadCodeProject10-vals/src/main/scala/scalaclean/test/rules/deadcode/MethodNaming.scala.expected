import scala.collection.immutable

//more a test that the names are unique and refer appropriately
package name1 {
  object A extends App {
    x.immutableSeq(Array(getClass))
  }
  object x {
    def immutableSeq(arr: Array[Class[_]]): immutable.Seq[Class[_]] = immutableSeq[Class[_]](arr)
  }
}
package name2 {
  object A extends App {
    x.immutableSeq(Array(1,2,3))
  }
  object x {

    def immutableSeq[T](arr: Array[T]): immutable.Seq[T] = if ((arr ne null) && arr.length > 0) Vector(arr: _*) else Nil
  }
}
